api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/TxPvlsDataset.java

```diff
@@ -0,0 +1,211 @@
+		ColumnParameters twtyTo24T = new ColumnParameters("20-24T", "20-24 years patients", "age=20-24");// 10
+		// columns for patients aged 25 to 29 years, defined by gender
+		ColumnParameters twty5To29M = new ColumnParameters("25-29M", "25-29 years males", "gender=M|age=25-29");// 11
+		ColumnParameters twty5To29F = new ColumnParameters("25-29F", "25-29 years female", "gender=F|age=25-29");// 12
+		ColumnParameters twty5To29T = new ColumnParameters("25-29T", "25-29 years patients", "age=25-29");// 13
+		// columns for patients aged 30 to 34 years, defined by gender
+		ColumnParameters thtyTo34M = new ColumnParameters("30-34M", "30-34 years males", "gender=M|age=30-34");// 14
+		ColumnParameters thtyTo34F = new ColumnParameters("30-34F", "30-34 years female", "gender=F|age=30-44");// 15
+		ColumnParameters thtyTo34T = new ColumnParameters("30-34T", "30-34 years patients", "age=30-34");// 16
+		// columns for patients aged 35 to 39 years, defined by gender
+		ColumnParameters thty5To39M = new ColumnParameters("35-39M", "35-39 years males", "gender=M|age=35-39");// 17
+		ColumnParameters thty5To39F = new ColumnParameters("35-39F", "35-39 years female", "gender=F|age=35-39");// 18
+		ColumnParameters thty5To39T = new ColumnParameters("35-39T", "35-39 years patients", "age=35-39");// 19
+		// columns for patients aged 40 to 44 years, defined by age
+		ColumnParameters ftyTo44M = new ColumnParameters("40-44M", "40-44 years males", "gender=M|age=40-44");// 19
+		ColumnParameters ftyTo44F = new ColumnParameters("40-44F", "40-44 years female", "gender=F|age=40-44");// 20
+		ColumnParameters ftyTo44T = new ColumnParameters("40-44T", "40-44 years patients", "age=40-44");// 21
+		// columns for patients aged 45 to 49 years, defined by age
+		ColumnParameters fty5To49M = new ColumnParameters("45-49M", "45-49 years males", "gender=M|age=45-49");// 22
+		ColumnParameters fty5To49F = new ColumnParameters("45-49F", "45-49 years female", "gender=F|age=45-49");// 23
+		ColumnParameters fty5To49T = new ColumnParameters("45-49T", "45-49 years patients", "age=45-49");// 24
+		// coloumn parameters for patients for 50 and above years, defined by gender
+		ColumnParameters fftyAndAboveM = new ColumnParameters(">=50M", ">=50 years males", "gender=M|age=>49");// 25
+		ColumnParameters fftyAndAboveF = new ColumnParameters(">=50F", ">=50 years female", "gender=F|age=>49");// 26
+		ColumnParameters fftyAndAboveT = new ColumnParameters(">=50T", ">=50 years patients", "age=>49");// 27
+		
+		return Arrays.asList(tenTo14M, tenTo14F, tenTo14T, fftnTo19M, fftnTo19F, fftnTo19T, twtyTo24M, twtyTo24F, twtyTo24T,
+		    twty5To29M, twty5To29F, twty5To29T, thtyTo34M, thtyTo34F, thtyTo34T, thty5To39M, thty5To39F, thty5To39T,
+		    ftyTo44M, ftyTo44F, ftyTo44T, fty5To49M, fty5To49F, fty5To49T, fftyAndAboveM, fftyAndAboveF, fftyAndAboveT);
+		
+	}
+	
+	private List<String> adultColumns() {
```

Comment:
It doesn't make sense to be defining an array with columnNames, when we already have a class which is meant to represent the column. Lets maybe move this to the ColumnProperties class.

Reply:
Hi @nthfloor I did not understand this, you mean I create another class called **ColumnProperties** and put this logic in? please shed more light

Reply:
Sorry, this is a typo, I meant to say *ColumnParameters* class.

Reply:
Since this is determined by the number of columns that our addRow method will display its data, it will vary for any other dataset. I will include them in the method. We can leave it as it is for now as we find a nicer way of including those into the nested columnParameter class

Reply:
Yes I realise the number of columns is determined by the dataset, so why not use the length of the array returned by `adultsDisagregation()` to determine the number of columns? 

You can add the column name to the *ColumnParameters* class and reference that in the `addRow()` function. Right now there are two arrays being passed into `addRow()` which represent the list of columns for that dataset - why not combine?

Reply:
Will apply as suggested. Done